mmf/models/mmf_transformer.py

```diff
@@ -0,0 +1,297 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+from copy import deepcopy
+from typing import Any, Dict, Type
+
+import torch
+from mmf.common.registry import registry
+from mmf.common.typings import DictConfig
+from mmf.models.transformers.base import (
+    BaseTransformer,
+    BaseTransformerConfigType,
+    BaseTransformerInput,
+)
+from mmf.modules.encoders import MultiModalEncoderBase
+from torch import Tensor, nn
+from transformers.modeling_bert import BertPooler, BertPredictionHeadTransform
+
+
+class ImageEncoder(MultiModalEncoderBase):
+    """Extends the MultiModalEncoderBase class which builds the encoder based on
+    the config parameters. We can set the type of image encoder(resnet50, resnet152,
+    resnext50 etc) and other parameters like num of features, type of pooling etc.
+    """
+
+    def __init__(self, config: DictConfig):
+        super().__init__(config)
+
+    def build(self):
+        self.encoder = self._build_modal_encoder(self.config.image_encoder)
+
+    def forward(self, x: Tensor) -> Tensor:
```

Comment:
I think this needs to be `Type[Tensor]` where Type is import from `typings`.

Reply:
No, `Tensor` type works as it is.